backend/src/main/java/org/example/restaurantmanagementapplication/rest/AuthController.java

```diff
@@ -45,13 +46,18 @@ public ResponseEntity<Authentication> login(@RequestBody LoginRequest loginReque
   public ResponseEntity<UserOut> me() {
     final var authentication = SecurityContextHolder.getContext().getAuthentication();
 
-    if (authentication == null || !authentication.isAuthenticated() || authentication.getPrincipal().equals("anonymousUser")) {
+    if (authentication == null || !authentication.isAuthenticated() || authentication.getPrincipal()
+        .equals("anonymousUser")) {
       return ResponseEntity.status(HttpStatus.UNAUTHORIZED).build();
     }
 
+    final var currentUser = getCurrentUser(authentication);
+    return ResponseEntity.ok().body(UserOut.fromEntity(currentUser));
+  }
+
+  private User getCurrentUser(Authentication authentication) {
     final var username = ((org.springframework.security.core.userdetails.User) authentication.getPrincipal()).getUsername();
     final var user = userService.findByUsername(username);
-
-    return ResponseEntity.ok().body(UserOut.fromEntity(user));
+    return user;
```

Comment:
Lines 60 - 61 could be replaced by an inline variable
```suggestion
    return userService.findByUsername(username);
```